### PR TITLE
Improving controller to use auto-generated code from code-generator

### LINKED ISSUE #20: Use code generators for custom resource
Currently we manually manage and create the k8s client and custom resources. However, it is becoming best practice to use the code generators for it.

This has a good introduction to the topic: https://blog.openshift.com/kubernetes-deep-dive-code-generation-customresources/

The code generators are here: https://github.com/kubernetes/code-generator

And a sample implementation can be looked at here: https://github.com/kubernetes/sample-controller

### PR DESCRIPTION
**Pull request by mheese**
This changes the controller implementation to use the `k8s.io/code-generator` to generate client, lister, informer and deepcopy functions.

Fixes #20 . 

**NOTE:** This will also need fixes in `trireme-kubernetes`.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -40,6 +40,18 @@
...
+  branch = "master"
+
+[[constraint]]
+  name = "k8s.io/gengo"
+  branch = "master"
+
+[[constraint]]
+  name = "k8s.io/kube-openapi"
+  branch = "master"
+
```
Comment (bvandewalle) [inline Gopkg.toml]: I'm actually wondering if we need to specify the branch ? It basically forces us to be on master.

File: main.go [hunk 2]
```diff
@@ -105,11 +115,26 @@ func buildConfig(kubeconfig string) (*rest.Config, error) {
...
 }
 
-func waitForSig() {
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT)
-	zap.L().Info("Everything started. Waiting for Stop signal")
-	// Waiting for a Sig
-	<-c
-	zap.L().Info("SIG received. Exiting")
+func createSignalChannel() <-chan struct{} {
```
Comment (bvandewalle) [inline main.go]: Thanks for refactoring this function.

We eventually need to move to the new `Context` I believe anyways, which would be used here eventually.

**Comment by bvandewalle** [comment]
Thanks for the cleanup @mheese . So we are now both supporting `v1alpha1` and `v1alpha2` ? wouldn't it be easier to deprecate completely and unsupport `v1alpha1` ?

**Comment by mheese** [comment]
@bvandewalle I was thinking of keeping it around more for testing upgrade paths in an upcoming issue (which I will create right now before I forget)

**Pull request merged**